contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,92 @@
+# VueJs
+
+Author: Lu Lechuan
+
+Vue.js is an open-source progressive [JavaScript framework](https://en.wikipedia.org/wiki/JavaScript_framework) for building user interfaces. It is designed to improve code quality and maintainablity.
+
+## Advantages of VueJs
+VueJs is an approachable, versatile and performant framework that helps to create a more maintainable and testable database.
+
+* Approachable: VueJs is very easy to learn. Comparing to other framework such as Augular, Vue is simple in term of API and design. Learning enough to build non-trivial applications typically takes less than a day.
+
+* Progressive: Vue is designed from the ground up to be incrementally adoptable. The core library is focused on the view layer only, and is easy to pick up and integrate with other libraries or existing projects. This means that if you have a large application, you can plug Vue into just a part of your application without disturbing the other components. A quote from Evan You - the owner of Vue is as follow:
+Vue.js is a more flexible, less opinionated solution ( than Angular ). That allows you to structure your app the way you want it to be, instead of being forced to do everything the Angular way. It’s only an interface layer so you can use it as a light feature in pages instead of a full blown SPA.
+
+* Versetile: Vue is perfectly capable of powering sophisticated single-page applications when used in combination with modern tooling and supporting libraries.
+
+* Clean: Vue symtax is simple and this can make the HTML pages very clean.
+
+* Other Advantages of VueJs: [What is Vue.js and What are its Advantages](https://hackernoon.com/what-is-vue-js-and-what-are-its-advantages-4071b7c7993d)
+
+* Detailed comparison of Vue with other JavaScript frameworks can be found from [Vue Guild: Comparison with Other Frameworks](https://vuejs.org/v2/guide/comparison.html).
+
+## Installation
+
+There are many ways to install VueJs, some of the following ways are mentioned below:
+* Import Vue in HTML `<script>` tag:
+  - [development version](https://vuejs.org/js/vue.js)
+  - [product version](https://vuejs.org/js/vue.min.js)
+* CDN:
+  - include the following into the html file:
+    <script src="https://cdn.jsdelivr.net/npm/vue@2.5.13/dist/vue.js"></script>
+* NPM:
+  - $ npm install vue
+* CLI:
+```cli
+    # install vue-cli
+    $ npm install --global vue-cli
+    # create a new project using the "webpack" template
+    $ vue init webpack my-project
+    # install dependencies and go!
+    $ cd my-project
+    $ npm install
+    $ npm run dev
```

Comment:
This part is a little messy to the reader because of all the `#` and `$` at the same indentation. Maybe you can put a space in between each command, or put the comments beside the command itself